### PR TITLE
Recommended pair for swap

### PR DESCRIPTION
**Pull request by amandeeprehal-lab49**
Recommended pair for swap

### CODE REVIEW

File: contracts/Factory.sol [hunk 1]
```diff
@@ -6,6 +6,8 @@ import "./ILPToken.sol";
 import "./common/IERC20.sol";
 import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
 import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "./Configuration.sol";
+import "hardhat/console.sol";
```
Comment (abhishek-k-sharma-lab49) [inline contracts/Factory.sol:10]: we can remove import "hardhat/console.sol";

**Review by shivamgarglab49 — CHANGES_REQUESTED** [review]

File: contracts/Factory.sol [hunk 2]
```diff
@@ -93,6 +108,81 @@ contract Factory is Initializable {
...
+        );
+
+        PairDetail[] memory recommendedPairs = findSwappedQtyForAllPairs(
+            feeItems,
+            _token0,
+            _token1,
+            _tokenToSwap,
+            _qtyToSwap
+        );
+        return findMaxQtyPool(recommendedPairs);
```
[the remark this answers is not in the document]
Comment (amandeeprehal-lab49) [reply contracts/Factory.sol:129]: You are right, let me fix.

File: contracts/Factory.sol [hunk 3]
```diff
@@ -93,6 +108,81 @@ contract Factory is Initializable {
...
+                    _slippage = pair.slippageOutGivenIn(_qtyToSwap);
+                } else {
+                    (, _qty, , ) = pair.getInGivenOut(_qtyToSwap);
+                    _token = _token1;
+                    _slippage = pair.slippageInGivenOut(_qtyToSwap);
+                }
+
+                recommendedPairs[count] = PairDetail(
+                    address(pair),
+                    _token0,
```
[the remark this answers is not in the document]
Comment (amandeeprehal-lab49) [reply contracts/Factory.sol:161]: Done

File: e2e-test/business/Factory.ts [hunk 4]
```diff
@@ -25,9 +26,11 @@ export default class Factory extends Base {
     client: Client = clientsInfo.operatorClient
   ) => {
     try {
+      const csDev = new ContractService();
       const args = new ContractFunctionParameters()
         .addAddress(this.htsAddress)
-        .addAddress(adminAddress);
+        .addAddress(adminAddress)
+        .addAddress(csDev.getContract(csDev.configuration).address);
```
[the remark this answers is not in the document]
Comment (amandeeprehal-lab49) [reply e2e-test/business/Factory.ts:33]: Done

File: contracts/Factory.sol (hunk 1 above)
Comment (amandeeprehal-lab49) [reply contracts/Factory.sol:10]: Removed

File: contracts/Factory.sol [hunk 5]
```diff
@@ -22,23 +24,36 @@ contract Factory is Initializable {
...
     modifier ifAdmin() {
         require(msg.sender == admin, "Factory: auth failed");
         _;
     }
 
     function setUpFactory(
         IBaseHTS _service,
-        address _admin
+        address _admin,
+        Configuration _configuration
```
[the remark this answers is not in the document]
Comment (amandeeprehal-lab49) [reply contracts/Factory.sol:50]: Configuration is a configuration, its fine if we are directly dependent on it. If we mock it, we will create exactly same mock that why I didn't mock. Basically its kind of global state, without any behaviour.

**Review by shivamgarglab49 — APPROVED** [review]

**Pull request merged**